source/llassetgen/include/llassetgen/Image.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <llassetgen/llassetgen_api.h>
+#include <llassetgen/llassetgen.h>
+
+#include <vector>
+#include <string>
+#include <memory>
+
+#include <png.h>
+
+#include <ft2build.h>
+#include FT_IMAGE_H
+
+namespace llassetgen {
+	class Image {
+		public:
+			LLASSETGEN_API Image(const FT_Bitmap &ft_bitmap);
+			LLASSETGEN_API Image(const std::string filepath);
+			LLASSETGEN_API Image(const size_t width, const size_t height, const size_t bit_depth);
+			LLASSETGEN_API void exportPng(const std::string filepath);
+			LLASSETGEN_API uint32_t at(const size_t x, const size_t y);
+			LLASSETGEN_API void put(const size_t x, const size_t y, const uint32_t data);
+			LLASSETGEN_API size_t get_width();
+			LLASSETGEN_API size_t get_height();
+			LLASSETGEN_API size_t get_bit_depth();
+			LLASSETGEN_API Image view(const size_t _min_x, const size_t _max_x, const size_t _min_y, const size_t _max_y);
+		private:
+			Image::Image(const size_t _min_x, const size_t _max_x, const size_t _min_y, const size_t _max_y, const uint8_t _bit_depth, const size_t _stride, const std::shared_ptr<uint8_t> _data);
```

Comment:
Unnecessary `const` in declaration (however, for the definition the `const` is fine).